components/director/internal/systemfetcher/service.go

```diff
@@ -149,6 +180,22 @@ func (s *SystemFetcher) saveSystemsForTenant(ctx context.Context, tenantMapping
 	ctx = persistence.SaveToContext(ctx, tx)
 
 	for _, system := range systems {
+		if system.AdditionalAttributes[LifecycleAttributeName] == LifecycleDeleted {
+			log.C(ctx).Infof("Getting system by name %s and system number %s", system.DisplayName, system.SystemNumber)
+			app, err := s.systemsService.GetByNameAndSystemNumber(ctx, system.DisplayName, system.SystemNumber)
+			if err != nil {
+				log.C(ctx).WithError(err).Errorf("Could not get system with name %s and system number %s", system.DisplayName, system.SystemNumber)
+				continue
+			}
+
+			if err := s.directorClient.DeleteSystemAsync(ctx, app.ID, tenantMapping.ID); err != nil {
```

Comment:
Not only can the `s.directorClient.DeleteSystemAsync` func try to gracefully handle concurrent operation errors, but we could even add an additional check here where if app.ready = false & deletedAt != nil => then don't make the `s.directorClient.DeleteSystemAsync` request as the system is already in deletion mode.

Also a log for a concurrent operation would be good

Reply:
An e2e test for this would be nice too